### DISCUSSION TITLE
Overriding the default path to the config file

### DISCUSSION THREAD

**Discussion by trojanmartin** [mburumaxwell/paklo#1280] [Ideas]
In current implementation azure devops extensions has hardcoded all possible paths to config file. 
What do you think about adding a configuration parameter that overrides the default path to the yaml configuration file ?

**Comment by mburumaxwell** [comment]
The idea is to keep this paths fixed so that intellisense works based on the schema and a well known path.
How do you go about that?

**Reply by trojanmartin** [reply]
It should only be used in some advanced use cases. The default behavior should remain the same (the default path should be used automatically, so InteliSense will work).

In our case, we use dependabot in a repository that contains many projects.  In case of a fixed path, it is possible to have only 1 pipeline (configuration file) and it runs too long when there are many projects present in the repository. 
The idea is to have multiple pipelines that can run at the same time and each pipeline will use a different configuration file.

**Reply by mburumaxwell** [reply]
When there are many repositories, it is better to use the [server version](https://github.com/tinglesoftware/dependabot-azure-devops/blob/main/docs/server.md) so that you do not clog your pipelines. Though it is yet to be updated to support directories and groups, it solves this more reliably.

**Reply by trojanmartin** [reply]
For many repositories, yes. What I'm trying to solve is the case where you have a monorepo with many projects. 
Also the configuration file can grow really fast in that situation. Being able to split it into smaller files might be useful.
